processor-api/src/main/java/com/synopsys/integration/alert/processor/api/NotificationProcessor.java

```diff
@@ -0,0 +1,91 @@
+package com.synopsys.integration.alert.processor.api;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.synopsys.integration.alert.common.enumeration.FrequencyType;
+import com.synopsys.integration.alert.common.enumeration.ProcessingType;
+import com.synopsys.integration.alert.common.persistence.model.job.DistributionJobModel;
+import com.synopsys.integration.alert.common.rest.model.AlertNotificationModel;
+import com.synopsys.integration.alert.processor.api.detail.NotificationDetailer;
+import com.synopsys.integration.alert.processor.api.detail.NotificationDetails;
+import com.synopsys.integration.alert.processor.api.digest.NotificationDigester;
+import com.synopsys.integration.alert.processor.api.distribute.NotificationDistributor;
+import com.synopsys.integration.alert.processor.api.extract.NotificationExtractor;
+import com.synopsys.integration.alert.processor.api.extract.model.ProviderMessage;
+import com.synopsys.integration.alert.processor.api.extract.model.SimpleMessage;
+import com.synopsys.integration.alert.processor.api.filter.FilterableNotificationWrapper;
+import com.synopsys.integration.alert.processor.api.filter.NotificationFilter;
+import com.synopsys.integration.alert.processor.api.summarize.NotificationSummarizer;
+
+public abstract class NotificationProcessor {
+    private final NotificationPreProcessor preProcessor;
+    private final NotificationFilter filter;
+    private final NotificationExtractor extractor;
+    private final NotificationDigester digester;
+    private final NotificationDetailer detailer;
+    private final NotificationSummarizer summarizer;
+    private final NotificationDistributor distributor;
+
+    protected NotificationProcessor(
+        NotificationPreProcessor preProcessor,
+        NotificationFilter filter,
+        NotificationExtractor extractor,
+        NotificationDigester digester,
+        NotificationDetailer detailer,
+        NotificationSummarizer summarizer,
+        NotificationDistributor distributor
+    ) {
+        this.preProcessor = preProcessor;
+        this.filter = filter;
+        this.extractor = extractor;
+        this.digester = digester;
+        this.detailer = detailer;
+        this.summarizer = summarizer;
+        this.distributor = distributor;
+    }
+
+    public abstract void processNotifications(List<AlertNotificationModel> notifications);
+
+    public abstract void processNotifications(FrequencyType frequency, List<AlertNotificationModel> notifications);
+
+    protected abstract Map<DistributionJobModel, List<FilterableNotificationWrapper<?>>> mapJobsToNotificationsOrSomething(List<? extends FilterableNotificationWrapper<?>> filterableNotifications);
+
+    // TODO include frequency?
```

Comment:
Frequency should be passed until filtering. The filter will filter contents based off of which frequency is being used. After that, frequency is no longer needed in this processing.

Reply:
Frequency is required when notifications come from the accumulator, but not when they come from resending an Audit (because we even want it resent to DAILY configs in that case. Any ideas about how to separate those concerns?

Reply:
Resending audits goes through the processor again? Regardless, we shouldn't even be running the processor unless it's explicitly told to run (i.e. Daily task, real time task, or audit runs it). This would mean we already fulfill the timing aspect of frequency, so now we just need to filter the jobs and be finished with frequency. So Frequency will always need to reach filtering imo

Reply:
To actually answer your question... Audit can kick off the processor twice passing each frequency

Reply:
Hmm. Not very efficient though.